Include/cpython/object.h

```diff
@@ -270,6 +270,9 @@ typedef struct _heaptypeobject {
     PyObject *ht_module;
     char *_ht_tpname;  // Storage for "tp_name"; see PyType_FromModuleAndSpec
     struct _specialization_cache _spec_cache; // For use by the specializer.
+#ifdef Py_GIL_DISABLED
+    Py_ssize_t _ht_id;  // ID used for thread-local refcounting
```

Comment:
This seems excessive if you are reusing them. Maybe use a 32 bit number?

Reply:
Also, `_ht_id` is not the clearest name. `unique_id` perhaps (since we know this is a heap type, the `ht` prefix adds little).

Reply:
More than 2^31-1 heap types seems unlikely, but not implausible. It also doesn't save any space, at least not for now, due to padding and alignment.